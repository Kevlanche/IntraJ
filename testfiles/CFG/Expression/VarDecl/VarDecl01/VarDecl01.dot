digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[6]"; "Opt[9]"; "List[10]"; "List[11]"}
{ rank = same; "List[7]"; "MethodDecl[12]"}
{ rank = same; "Modifier[8]"; "Entry[13]"; "Modifiers[14]"; "PrimitiveTypeAccess[16]"; "List[17]"; "List[18]"; "Opt[19]"; "Exit[20]"}
{ rank = same; "List[15]"; "Block[20]"}
{ rank = same; "List[21]"}
{ rank = same; "VarDeclStmt[22]"}
{ rank = same; "Modifiers[23]"; "PrimitiveTypeAccess[25]"; "List[26]"}
{ rank = same; "List[24]"; "VariableDeclarator[27]"}
{ rank = same; "List[28]"; "Opt[29]"}
{ rank = same; "IntegerLiteral[30]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < VarDecl01 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[6]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[7]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[8]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[9]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[10]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[11]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodDecl[12]" [label="MethodDecl\n < m >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[13]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[14]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[15]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[16]" [label="PrimitiveTypeAccess\n < void >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[17]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[18]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[19]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[20]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[21]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[22]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[23]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[24]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[25]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[27]" [label="VariableDeclarator\n < int a >", fillcolor=white   style=filled  ]
"List[28]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[29]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[30]" [label="IntegerLiteral\n < 0 >", fillcolor=white   style=filled  ]
"Exit[20]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[7]" -> "Modifier[8]" [style=dashed, color=gray]
"Modifiers[6]" -> "List[7]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[9]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[10]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Entry[13]" [style=dashed, color=gray]
"Modifiers[14]" -> "List[15]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Modifiers[14]" [style=dashed, color=gray]
"MethodDecl[12]" -> "PrimitiveTypeAccess[16]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[17]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[18]" [style=dashed, color=gray]
"Modifiers[23]" -> "List[24]" [style=dashed, color=gray]
"VarDeclStmt[22]" -> "Modifiers[23]" [style=dashed, color=gray]
"VarDeclStmt[22]" -> "PrimitiveTypeAccess[25]" [style=dashed, color=gray]
"VariableDeclarator[27]" -> "List[28]" [style=dashed, color=gray]
"Opt[29]" -> "IntegerLiteral[30]" [style=dashed, color=gray]
"VariableDeclarator[27]" -> "Opt[29]" [style=dashed, color=gray]
"List[26]" -> "VariableDeclarator[27]" [style=dashed, color=gray]
"VarDeclStmt[22]" -> "List[26]" [style=dashed, color=gray]
"List[21]" -> "VarDeclStmt[22]" [style=dashed, color=gray]
"Block[20]" -> "List[21]" [style=dashed, color=gray]
"Opt[19]" -> "Block[20]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Opt[19]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Exit[20]" [style=dashed, color=gray]
"List[11]" -> "MethodDecl[12]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[11]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[20]" -> "VariableDeclarator[27]" [color=red, constraint=false]
"VariableDeclarator[27]" -> "Exit[20]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[27]" -> "IntegerLiteral[30]" [color=red, constraint=false]
"IntegerLiteral[30]" -> "VariableDeclarator[27]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[30]" -> "Entry[13]" [color=red, constraint=false]
"Entry[13]" -> "IntegerLiteral[30]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"Modifiers[6]" -> "Opt[9]" -> "List[10]" -> "List[11]" [style=invis]
"Entry[13]" -> "Modifiers[14]" -> "PrimitiveTypeAccess[16]" -> "List[17]" -> "List[18]" -> "Opt[19]" -> "Exit[20]" [style=invis]
"Modifiers[23]" -> "PrimitiveTypeAccess[25]" -> "List[26]" [style=invis]
"List[28]" -> "Opt[29]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
32", fillcolor=1]
Edges [label="Total Nr. Edges
37", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
4", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
3", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}