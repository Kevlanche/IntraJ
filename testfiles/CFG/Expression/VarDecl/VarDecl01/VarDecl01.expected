NodeName: Entry[13] Successors: [IntegerLiteral[30]]
NodeName: IntegerLiteral[30] Successors: [VariableDeclarator[27]]
NodeName: VariableDeclarator[27] Successors: [Exit[20]]
NodeName: Exit[20] Successors: []

NodeName: Exit[20] Predecessor: [VariableDeclarator[27]]
NodeName: VariableDeclarator[27] Predecessor: [IntegerLiteral[30]]
NodeName: IntegerLiteral[30] Predecessor: [Entry[13]]
NodeName: Entry[13] Predecessor: []
